xarray/core/dtypes.py

```diff
@@ -60,22 +61,22 @@ def maybe_promote(dtype: np.dtype) -> tuple[np.dtype, Any]:
     # N.B. these casting rules should match pandas
     dtype_: np.typing.DTypeLike
     fill_value: Any
-    if np.issubdtype(dtype, np.floating):
+    if isdtype(dtype, "real floating"):
```

Comment:
The chances of typos sneaking in here are quite large.

Shall we use a `DType` Enum instead?

Reply:
this is what `np.isdtype` and the array API's `isdtype` are using, so I just copied that. We can, of course, use enums, but then we'd have to be able to cast to these strings.

Another option would be to modify the translation logic to not use `dict.get(kind, kind)`, but instead raise a descriptive error.

Reply:
One compromise would be to define top level constants in this module, e.g.,
```python
REAL_FLOATING = 'real_floating
```

Reply:
In the most recent commit I went with the `enum` option (since the access is at runtime I don't think it makes too much of a difference, now that we actually check for valid dtype strings). I'm not sure I like it better than the strings, though... in any case, if you find it easier to read I can also make use of this in `xarray.core.duck_array_ops`.

Reply:
I removed the `enum` again in the interest of being as small of a wrapper around `xp.isdtype` as possible, and the error is either raised by our own version of `numpy.isdtype`, or by whatever implementation of `isdtype` we're dispatching to.